cognite/client/data_classes/sequences.py

```diff
@@ -461,36 +615,115 @@ def column_external_ids(self) -> list[str]:
+            rows=rows_parsed,
+        )
 
 
-class SequenceDataList(CogniteResourceList[SequenceData]):
-    _RESOURCE = SequenceData
+class SequenceRowsList(CogniteResourceList[SequenceRows]):
+    _RESOURCE = SequenceRows
 
     def __str__(self) -> str:
         return json.dumps(self.dump(), indent=4)
 
-    def to_pandas(self, column_names: str = "externalId|columnExternalId") -> pandas.DataFrame:  # type: ignore[override]
+    @overload  # type: ignore[override]
+    def to_pandas(
+        self,
+        key: Literal["id", "external_id"] = "external_id",
+        column_names: ColumnNames = "externalId|columnExternalId",
+        concat: Literal[True] = True,
+    ) -> pandas.DataFrame:
+        ...
+
+    @overload
+    def to_pandas(
+        self,
+        key: Literal["external_id"] = "external_id",
+        column_names: ColumnNames = "externalId|columnExternalId",
+        concat: Literal[False] = False,
+    ) -> dict[str, pandas.DataFrame]:
+        ...
+
+    @overload
+    def to_pandas(
+        self,
+        key: Literal["id"],
+        column_names: ColumnNames = "externalId|columnExternalId",
+        concat: Literal[False] = False,
+    ) -> dict[int, pandas.DataFrame]:
+        ...
+
+    def to_pandas(
+        self,
+        key: Literal["id", "external_id"] = "external_id",
+        column_names: ColumnNames = "externalId|columnExternalId",
+        concat: bool = False,
+    ) -> pandas.DataFrame | dict[str, pandas.DataFrame] | dict[int, pandas.DataFrame]:
         """Convert the sequence data list into a pandas DataFrame. Each column will be a sequence.
 
         Args:
-            column_names (str):  Which field to use as column header. Can use any combination of "externalId", "columnExternalId", "id" and other characters as a template.
+            key (Literal["id", "external_id"]): If concat = False, this decides which field to use as key in the dictionary. Defaults to "external_id".
+            column_names (ColumnNames): Which field to use as column header. Can use any combination of "externalId", "columnExternalId", "id" and other characters as a template.
+            concat (bool): Whether to concatenate the sequences into a single DataFrame or return a dictionary of DataFrames. Defaults to False.
 
         Returns:
-            pandas.DataFrame: The sequence data list as a pandas DataFrame.
+            pandas.DataFrame | dict[str, pandas.DataFrame] | dict[int, pandas.DataFrame]: The sequence data list as a pandas DataFrame.
         """
         pd = local_import("pandas")
-        return pd.concat([seq_data.to_pandas(column_names=column_names) for seq_data in self.data], axis=1)
+        if concat:
+            return pd.concat([seq.to_pandas(column_names) for seq in self], axis=1)
+
+        if key == "external_id":
+            return {seq.external_id: seq.to_pandas(column_names) for seq in self}
```

Comment:
Should probably have a check here ensuring that all xids exist (if not, all with missing would fight for the same `None` key in the `dict`)